pkg/collector/corechecks/network-devices/cisco-sdwan/payload/cedge_interface.go

```diff
@@ -0,0 +1,126 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+package payload
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/network-devices/cisco-sdwan/client"
+	devicemetadata "github.com/DataDog/datadog-agent/pkg/networkdevice/metadata"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// https://github.com/YangModels/yang/blob/9442dda17a9a5f1f0db548512446e3d9ca37a955/vendor/cisco/xe/17131/Cisco-IOS-XE-interfaces-oper.yang#L305
+var cEdgeOperStatusMap = map[string]devicemetadata.IfOperStatus{
+	"if-oper-state-down":             devicemetadata.OperStatusDown,
+	"if-oper-state-invalid":          devicemetadata.OperStatusDown,
+	"if-oper-state-ready":            devicemetadata.OperStatusUp,
+	"if-oper-state-no-pass":          devicemetadata.OperStatusDown,
+	"if-oper-state-test":             devicemetadata.OperStatusTesting,
+	"if-oper-state-unknown":          devicemetadata.OperStatusUnknown,
+	"if-oper-state-dormant":          devicemetadata.OperStatusDormant,
+	"if-oper-state-not-present":      devicemetadata.OperStatusNotPresent,
+	"if-oper-state-lower-layer-down": devicemetadata.OperStatusLowerLayerDown,
+}
+
+// https://github.com/YangModels/yang/blob/9442dda17a9a5f1f0db548512446e3d9ca37a955/vendor/cisco/xe/17131/Cisco-IOS-XE-interfaces-oper.yang#L212
+var cEdgeAdminStatusMap = map[string]devicemetadata.IfAdminStatus{
+	"if-state-unknown": devicemetadata.AdminStatusDown,
+	"if-state-up":      devicemetadata.AdminStatusUp,
+	"if-state-down":    devicemetadata.AdminStatusDown,
+	"if-state-test":    devicemetadata.AdminStatusTesting,
+}
+
+// CEdgeInterface is an implementation of CiscoInterface for cEdge devices
+type CEdgeInterface struct {
+	client.CEdgeInterfaceState
+}
+
+// ID returns an unique interface ID
+func (itf *CEdgeInterface) ID() string {
+	return fmt.Sprintf("%s:%s", itf.VmanageSystemIP, itf.Ifname)
+}
+
+// Index returns the interface index
+func (itf *CEdgeInterface) Index() int {
+	index, err := strconv.Atoi(itf.Ifindex)
+	if err != nil {
+		log.Warnf("Unable to parse cEdge interface %s index %s", itf.Ifname, itf.Ifindex)
+	}
+	return index
```

Comment:
If we return 0 here, I believe we should instead return an error in this function

Reply:
Addressed in 9347cb0